client/state/site-settings/exporter/actions.js

```diff
@@ -83,28 +96,32 @@ export function advancedSettingsFail( siteId, error ) {
  * @param  {Number}   siteId  The ID of the site to export
  * @return {Function}         Action thunk
  */
-export function startExport( siteId ) {
-	return ( dispatch ) => {
+export function startExport( siteId, exportAll = true ) {
+	return ( dispatch, getState ) => {
 		if ( ! siteId ) {
 			return;
 		}
 
 		dispatch( {
 			type: EXPORT_START_REQUEST,
-			siteId: siteId
+			siteId,
+			exportAll,
 		} );
 
+		const state = getState();
+		const advancedSettings = exportAll ? null : prepareExportRequest( state, siteId );
+
 		const success =
 			() => dispatch( exportStarted( siteId ) );
 
 		const failure =
 			error => dispatch( exportFailed( siteId, error ) );
 
 		return wpcom.undocumented()
-			.startExport( siteId )
+			.startExport( siteId, advancedSettings )
 			.then( success )
```

Comment:
Use your best judgement, we had some back and forth on mapping before. It makes sense when the API is very far off from what you need, but it also adds maintenance overhead. Some PHP vs JS naming can be abstracted with QueryComponents and Selectors. See: https://github.com/Automattic/wp-calypso/pull/5251#discussion_r62325558